Source/hydro/Castro_mol_hydro.cpp

```diff
@@ -439,62 +439,60 @@ Castro::construct_mol_hydro_source(Real time, Real dt, MultiFab& A_update)
 
           const Box& tbx = amrex::grow(bx, 2);
 
-          qm.resize(tbx, NQ*AMREX_SPACEDIM);
+          qm.resize(tbx, NQ);
           Elixir elix_qm = qm.elixir();
 
-          qp.resize(tbx, NQ*AMREX_SPACEDIM);
+          qp.resize(tbx, NQ);
           Elixir elix_qp = qp.elixir();
 
-          if (do_hydro) {
-
-            if (ppm_type == 0) {
+          // compute the fluxes and add artificial viscosity
 
-              dq.resize(obx, NQ);
-              Elixir elix_dq = dq.elixir();
+          q_int.resize(obx, NQ);
+          Elixir elix_q_int = q_int.elixir();
 
-#pragma gpu box(obx)
-              ca_mol_plm_reconstruct
-                (AMREX_INT_ANYD(obx.loVect()), AMREX_INT_ANYD(obx.hiVect()),
-                 BL_TO_FORTRAN_ANYD(q[mfi]),
-                 BL_TO_FORTRAN_ANYD(flatn),
-                 BL_TO_FORTRAN_ANYD(dq),
-                 BL_TO_FORTRAN_ANYD(qm),
-                 BL_TO_FORTRAN_ANYD(qp),
-                 AMREX_REAL_ANYD(dx),
-                 AMREX_INT_ANYD(domain_lo), AMREX_INT_ANYD(domain_hi));
+          for (int idir = 0; idir < AMREX_SPACEDIM; ++idir) {
 
-            } else {
+            const int idir_f = idir + 1;
 
-#pragma gpu box(obx)
-              ca_mol_ppm_reconstruct
-                (AMREX_INT_ANYD(obx.loVect()), AMREX_INT_ANYD(obx.hiVect()),
-                 BL_TO_FORTRAN_ANYD(q[mfi]),
-                 BL_TO_FORTRAN_ANYD(flatn),
-                 BL_TO_FORTRAN_ANYD(qm),
-                 BL_TO_FORTRAN_ANYD(qp),
-                 AMREX_REAL_ANYD(dx));
-            }
+            if (do_hydro) {
 
-          }
+              if (ppm_type == 0) {
 
-          // compute the fluxes and add artificial viscosity
+                dq.resize(obx, NQ);
+                Elixir elix_dq = dq.elixir();
```

Comment:
This won't work, but we can resolve that when we do the final pass for GPUs.

Reply:
okay, but can you briefly share why it doesn't work?